auth/src/main/java/org/aerogear/auth/RoleKey.java

```diff
@@ -0,0 +1,24 @@
+package org.aerogear.auth;
+
+/**
+ * Created by rachael on 25/01/2018.
+ */
+
+public class RoleKey {
+
+    private String roleName;
+    private RoleType roleType;
+
+    public RoleKey(IRole role) {
+        roleName = role.getRoleName();
+        roleType = role instanceof RealmRole ? RoleType.REALM : RoleType.CLIENT;
```

Comment:
Instead of checking object class, wouldn't be better to add a 'RoleType getType()' method in the `IRole` interface and implement it in each role class, returning the appropriate enum?